batchflow/models/eager_torch/encoder_decoder.py

```diff
@@ -0,0 +1,504 @@
+""" Encoder, decoder, encoder-decoder architectures. """
+import torch
+import torch.nn as nn
+
+from .base import EagerTorch
+from .utils import get_shape
+from .layers import ConvBlock, Upsample, Combine, Crop
+from ..utils import unpack_args
+
+
+
+class EncoderModule(nn.Module):
+    """ Encoder: create compressed representation of an input by reducing its spatial dimensions. """
+    def __init__(self, inputs=None, return_all=True, **kwargs):
+        super().__init__()
+        self.return_all = return_all
+        self._make_modules(inputs, **kwargs)
+
+    def forward(self, x):
+        b_counter, d_counter = 0, 0
+        outputs = []
+
+        for _ in range(self.num_stages):
+            for letter in self.encoder_layout:
+                if letter in ['b']:
+                    x = self.encoder_b[b_counter](x)
```

Comment:
do you really need different lists `self.encoder_b` and `self.encoder_d`? Why not single `self.ops` list?